query/src/main/java/org/infinispan/query/MassIndexer.java

```diff
@@ -20,14 +21,14 @@
    /**
     * Deletes all the indexes and skip the reindexing.
     */
-   CompletableFuture<Void> purge();
+   CompletionStage<Void> purge();
```

Comment:
This changes the API of the MassIndexer. I am deprecating it in favour of another interface called ```Indexer``` in https://github.com/infinispan/infinispan/pull/8303, that has only non-blocking methods that return ```CompletionStage```.
I think the old interface should not be changed though.

Reply:
Actually, don't worry, I will revert it back in my PR

Reply:
I missed Gustavo's comments and wrote the same thing :)